### PR TITLE
Widen CallTreeWidget's "Thread / Function" column on row expansion

### PR DESCRIPTION
**Pull request by dpallotti**
When expanding a row of the top-down or bottom-up view, the newly visible nodes
could be completely hidden if the "Thread / Function" column was too narrow.
We now possibly widen the column so that the arrow and the first few characters
of these nodes are visible.

Keeping performance in mind, we have to distinguish between the case of a single
row expanded manually, and the case of many rows expanded programmatically with
a context menu action or with the search functionality.

Bug: http://b/218449940

Test: Multiple manual tests expanding rows manually, with the context menu
actions, and with the search bar.

### CODE REVIEW

File: src/OrbitQt/CallTreeWidget.h [hunk 1]
```diff
@@ -133,6 +138,7 @@ class CallTreeWidget : public QWidget {
     kDone = 2            // The columns have been resized (once)
   };
   ColumnResizingState column_resizing_state_ = ColumnResizingState::kInitial;
+  bool resize_thread_or_function_column_on_row_expanded_ = true;
```
Comment (karupayun) [inline src/OrbitQt/CallTreeWidget.h:141]: The usage of this variable is a bit strange to me but I guess is the best way to solve this problem with signals.

File: src/OrbitQt/CallTreeWidget.cpp [hunk 2]
```diff
@@ -184,6 +185,74 @@ void CallTreeWidget::ResizeColumnsIfNecessary() {
   column_resizing_state_ = ColumnResizingState::kDone;
 }
 
+static int FindDepthDifferenceFromIndexToDeepestVisibleNode(QTreeView* tree_view,
```
Comment (karupayun) [inline src/OrbitQt/CallTreeWidget.cpp:188]: Nit: Too long name, how about? `FindHeightOfVisibleNodesSubTree `

**Review by karupayun — APPROVED** [review]
LGTM! Thanks for fixing this. I like the solution.

File: src/OrbitQt/CallTreeWidget.h (hunk 1 above)
Comment (dpallotti) [reply src/OrbitQt/CallTreeWidget.h:141]: Yeah, I don't like it that much either, but something like this is somewhat necessary as the performance difference is measurable.
By the way, a not-so-different alternative would be to not use this field but call `disconnect` and `connect`. I went for the variable as in theory `OnRowExpanded` could do something else that is cheap and needs to be done at all times. But the general principle is the same.

File: src/OrbitQt/CallTreeWidget.cpp (hunk 2 above)
Comment (dpallotti) [reply src/OrbitQt/CallTreeWidget.cpp]: Long names are not a problem (E.g., style guide: "Do not worry about saving horizontal space as it is far more important to make your code immediately understandable by a new reader."). However, I like expressing the concept like that: I changed to `ComputeHeightOfSubtreeOfVisibleNodes`.

**Review by antonrohr — APPROVED** [review]

File: src/OrbitQt/CallTreeWidget.cpp (hunk 2 above)
Comment (karupayun) [reply src/OrbitQt/CallTreeWidget.cpp]: Yeah, I wanted to point out to Long names that can be nicer being short. I like yours!

**Review by beckerhe — APPROVED** [review]
Nice. LGTM.

**Pull request merged**